pkg/operator/events/recorder.go

```diff
@@ -34,28 +35,34 @@ type recorder struct {
 }
 
 // Event emits the normal type event and allow formatting of message.
-func (r *recorder) Eventf(reason, messageFmt string, args ...interface{}) error {
-	return r.Event(reason, fmt.Sprintf(messageFmt, args...))
+func (r *recorder) Eventf(reason, messageFmt string, args ...interface{}) {
+	r.Event(reason, fmt.Sprintf(messageFmt, args...))
 }
 
 // Warning emits the warning type event and allow formatting of message.
-func (r *recorder) Warningf(reason, messageFmt string, args ...interface{}) error {
-	return r.Warning(reason, fmt.Sprintf(messageFmt, args...))
+func (r *recorder) Warningf(reason, messageFmt string, args ...interface{}) {
+	r.Warning(reason, fmt.Sprintf(messageFmt, args...))
 }
 
 // Event emits the normal type event.
-func (r *recorder) Event(reason, message string) error {
-	_, err := r.eventClient.Create(r.makeEvent(r.involvedObjectRef, corev1.EventTypeNormal, reason, message))
-	return err
+func (r *recorder) Event(reason, message string) {
+	event := makeEvent(r.involvedObjectRef, r.sourceComponent, corev1.EventTypeNormal, reason, message)
```

Comment:
return early if r is nil

Reply:
how can be 'r' nil ?

Reply:
you use the Recorder interface in the plumbing. So yes, it can't be nil here, but the interface can.